src/client/connection.rs

```diff
@@ -388,70 +369,10 @@ impl Connection {
         Ok(())
     }
 
-    #[cfg(not(windows))]
-    async fn find_tcp_port(addr: SocketAddr, _: &str) -> crate::Result<SocketAddr> {
-        Ok(addr)
-    }
-
-    /// Use the SQL Browser to find the correct TCP port for the server
-    /// instance.
-    #[cfg(windows)]
-    async fn find_tcp_port(mut addr: SocketAddr, instance_name: &str) -> crate::Result<SocketAddr> {
-        // First resolve the instance to a port via the
-        // SSRP protocol/MS-SQLR protocol [1]
-        // [1] https://msdn.microsoft.com/en-us/library/cc219703.aspx
-
-        let local_bind: SocketAddr = if addr.is_ipv4() {
-            "0.0.0.0:0".parse().unwrap()
-        } else {
-            "[::]:0".parse().unwrap()
-        };
-
-        let msg = [&[4u8], instance_name.as_bytes()].concat();
-
-        let mut socket = UdpSocket::bind(&local_bind).await?;
-        socket.send_to(&msg, &addr).await?;
-
-        let mut buf = vec![0u8; 4096];
-        let timeout = Duration::from_millis(1000);
-
-        let len = time::timeout(timeout, socket.recv(&mut buf))
-            .await
-            .map_err(|_: time::Elapsed| {
-                Error::Conversion(
-                    format!(
-                        "SQL browser timeout during resolving instance {}",
-                        instance_name
-                    )
-                    .into(),
-                )
-            })??;
-
-        buf.truncate(len);
-
-        let err = Error::Conversion(
-            format!("Could not resolve SQL browser instance {}", instance_name).into(),
-        );
-
-        if len == 0 {
-            return Err(err);
-        }
-
-        let response = str::from_utf8(&buf[3..len])?;
-
-        let port: u16 = response
-            .find("tcp;")
-            .and_then(|pos| response[pos..].split(';').nth(1))
-            .ok_or(err)
-            .and_then(|val| Ok(val.parse()?))?;
-
-        addr.set_port(port);
-
-        Ok(addr)
-    }
 }
 
-impl Stream for Connection {
+
+impl<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> Stream for Connection<S> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.